src/inference/src/system_conf.cpp

```diff
@@ -185,7 +185,7 @@ void set_cpu_used(const std::vector<int>& cpu_ids, const int used) {}
 
 #else
 
-static CPU cpu;
+static thread_local CPU cpu;
```

Comment:
is not construction of static variable thread-safe?

Reply:
Multiple threads are accessing and changing this object (which is basically shared between the threads) after it is constructed.
This is just a quick fix, in general we better to reconsider the implementation and not use static object at all.

Reply:
> Should we just use `thread_local CPU cpu;`?

no, we don't need to expose this variable to other translation units.